src/courtroom.cpp

```diff
@@ -2367,19 +2367,34 @@ void Courtroom::display_pair_character(QString other_charid, QString other_offse
       ui_vp_sideplayer_char->move(ui_viewport->width() * offset_x / 100,
                                   ui_viewport->height() * offset_y / 100);
 
-  QString customchar;
-  if (ao_app->is_customchat_enabled())
-    customchar = m_chatmessage[CHAR_NAME];
-  if (ui_vp_showname->text().trimmed().isEmpty()) // Whitespace showname
-  {
-    QString chatbox = ao_app->get_chat(customchar);
-    QString chatbox_path = ao_app->get_theme_path("chatblank");
-    if (chatbox != "") {
-      chatbox_path = ao_app->get_theme_path("misc/" + chatbox + "/chatblank");
-      if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
-        chatbox_path =
-            ao_app->get_base_path() + "misc/" + chatbox + "/chatblank";
-        if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
+      QString customchar;
+      if (ao_app->is_customchat_enabled())
+        customchar = m_chatmessage[CHAR_NAME];
+      if (ui_vp_showname->text().trimmed().isEmpty()) // Whitespace showname
+      {
+        QString chatbox = ao_app->get_chat(customchar);
+        QString chatbox_path = ao_app->get_theme_path("chatblank");
+        if (chatbox != "") {
+          chatbox_path = ao_app->get_theme_path("misc/" + chatbox + "/chatblank");
+          if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
+            chatbox_path =
+              ao_app->get_base_path() + "misc/" + chatbox + "/chatblank";
+            if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
+              chatbox_path = ao_app->get_theme_path("chat");
+              if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
+                chatbox_path = ao_app->get_theme_path("chatbox");
+                if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
+                  chatbox_path = ao_app->get_default_theme_path("chat");
+                  if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
+                    chatbox_path = ao_app->get_default_theme_path("chatbox");
+                    ui_vp_chatbox->set_chatbox(chatbox_path);
+                  }
+                }
+              }
+            }
+          }
+        }
+        else if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
           chatbox_path = ao_app->get_theme_path("chat");
           if (!ui_vp_chatbox->set_chatbox(chatbox_path)) {
             chatbox_path = ao_app->get_theme_path("chatbox");
```

Comment:
This is cursed

Reply:
agreed however this particular bit of cursedness was there in #302

Reply:
i will probably make this a for loop once the merge conflicts are finished conflicting